src/features/broker-protection/actions/extract.js

```diff
@@ -278,114 +274,46 @@ export function aggregateFields (profile) {
-            }
-
-            const profileUrl = Array.isArray(elementValue) ? elementValue[0] : elementValue
-            profile.identifier = getIdFromProfileUrl(profileUrl, value.identifierType, value.identifier)
-            return profile
-        }
+export function extractValue (outputFieldKey, extractorParams, elementValues) {
+    switch (outputFieldKey) {
+    case 'age': return new AgeExtractor().extract(elementValues, extractorParams)
+    case 'name': return new NameExtractor().extract(elementValues, extractorParams)
+
+    // all addresses are processed the same way
+    case 'addressFull':
+    case 'addressFullList':
+        return new AddressFullExtractor().extract(elementValues, extractorParams)
+    case 'addressCityState':
+    case 'addressCityStateList':
+        return new CityStateExtractor().extract(elementValues, extractorParams)
+
+    case 'alternativeNamesList': return new AlternativeNamesExtractor().extract(elementValues, extractorParams)
+    case 'relativesList': return new RelativesExtractor().extract(elementValues, extractorParams)
+    case 'phone':
+    case 'phoneList':
+        return new PhoneExtractor().extract(elementValues, extractorParams)
+    case 'profileUrl': return new ProfileUrlExtractor().extract(elementValues, extractorParams)
     }
-
-    if (key in extractors) {
-        return extractors[key]()
-    }
-
     return null
 }
 
 /**
- * @param {string|any[]} inputList
+ * @param {string} inputList
  * @param {string} [separator]
  * @return {string[]}
  */
 export function stringToList (inputList, separator) {
-    // if the list is already an array then we can return the list
-    if (Array.isArray(inputList)) return inputList
-    if (inputList === '') return []
-
-    if (separator) {
-        return inputList
-            .split(separator)
-            .map(item => item.trim())
-            .filter(Boolean)
-    }
-
-    return inputList
-        .split(/[|\n•·]/)
-        .map(item => item.trim())
-        .filter(Boolean)
-}
-
-/**
- * @param {string[]} inputList
- * @return {{ city: string, state: string|null }[] }
- */
-export function getCityStateCombos (inputList) {
-    const output = []
-    for (let item of inputList) {
-        let words
-        // Strip out the zip code since we're only interested in city/state here.
-        item = item.replace(/,?\s*\d{5}(-\d{4})?/, '')
-
-        if (item.includes(',')) {
-            words = item.split(',').map(item => item.trim())
-        } else {
-            words = item.split(' ').map(item => item.trim())
-        }
-        // we are removing this partial city/state combos at the end (i.e. Chi...)
-        if (words.length === 1) { continue }
-
-        const state = words.pop()
-        const city = words.join(' ')
-
-        output.push({ city, state: state || null })
-    }
-    return output
+    const defaultSeparator = /[|\n•·]/
+    return cleanArray(inputList.split(separator || defaultSeparator))
 }
```

Comment:
in the end, this function was just 'split with a separator or default separator'